client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -92,18 +163,70 @@ class SpeedUpSiteSettings extends Component {
 	}
 }
 
-export default connect( state => {
-	const selectedSiteId = getSelectedSiteId( state );
-	const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
-	const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
-		state,
-		selectedSiteId,
-		'photon'
-	);
+export default connect(
+	state => {
+		const selectedSiteId = getSelectedSiteId( state );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			selectedSiteId,
+			'photon'
+		);
+		const photonModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon' );
+		const assetCdnModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon-cdn' );
+		const isPhotonActivating = isActivatingJetpackModule( state, selectedSiteId, 'photon' );
+		const isAssetCdnActivating = isActivatingJetpackModule( state, selectedSiteId, 'photon-cdn' );
+		const isPhotonDeactivating = isDeactivatingJetpackModule( state, selectedSiteId, 'photon' );
+		const isAssetCdnDeactivating = isDeactivatingJetpackModule(
+			state,
+			selectedSiteId,
+			'photon-cdn'
+		);
 
-	return {
-		photonModuleUnavailable: siteInDevMode && moduleUnavailableInDevMode,
-		selectedSiteId,
-		siteSlug: getSiteSlug( state, selectedSiteId ),
-	};
-} )( localize( SpeedUpSiteSettings ) );
+		let togglingSiteAccelerator;
+		// First Photon activating.
+		if ( isPhotonActivating ) {
```

Comment:
We don't seem to need this deep level of specificity here. Instead I'd go with:

```
const togglingSiteAccelerator = (
	isAssetCdnActivating ||
	isAssetCdnDeactivating ||
	isPhotonActivating ||
	isPhotonDeactivating
);
```

which should be good enough once we make sure to disable all the UI together.

Reply:
That definitely makes things simpler. I took care of that.

Reply:
I've reverted that change. The specificity is needed or you run into issues in specific cases, as I tried to explained in 8fb7f287a60a09b9a8fcfae625a3b2af7a1074c6